### PR TITLE
railway: init at 1.8.3

### PR DESCRIPTION
**Pull request by Craftzman7**
###### Description of changes

Add the Railway CLI to Nixpkgs.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/railway/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  meta = with lib; {
+    description = "Railway CLI";
+    homepage = "https://railway.app";
+    license = licenses.mit;
+    maintainers = with maintainers; [ Crafter ];
+  };
+}
+
+
```
Comment (winterqt) [inline pkgs/development/tools/railway/default.nix:28]: Remove these stray newlines.

**Review by winterqt — CHANGES_REQUESTED** [review]
Please split your changes into two commits:
1. adding yourself as a maintainer
2. adding the package

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please put the maintainer change into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

Also please squash the railway init commits together.

**Comment by winterqt** [comment]
> Please put the maintainer change into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

Looks like it is already, no? It's in between `cpu` and `craigem`.

> Also please squash the railway init commits together.

I figured that was implied by my comment to split the changes into 2 commits.

**Comment by winterqt** [comment]
It's correct:

> <SuperSandro2000> that sorting is in my reply template by default and I didn't mind to take it out

**Review by winterqt — CHANGES_REQUESTED** [review]
Can you test with `ldflags = [ "-s" "-w" ]`, and if it builds and runs fine, push that? LGTM otherwise.

**Review by winterqt — APPROVED** [review]

**Pull request merged**